v2/tools/generator/internal/reporting/structure_report.go

```diff
@@ -16,32 +17,32 @@ const (
 	lastBlockIndent = "    "
 )
 
-// debugReport represents a heirarchical dump of debug information
-type debugReport struct {
+// StructureReport represents a hierarchical dump of structural information
+type StructureReport struct {
 	line   string
-	nested []*debugReport
+	nested []*StructureReport
 }
 
-// newDebugReport creates a new debugReport
-func newDebugReport(line string) *debugReport {
-	return &debugReport{line: line}
+// NewStructureReport creates a new StructureReport
+func NewStructureReport(line string) *StructureReport {
+	return &StructureReport{line: line}
 }
 
-// add adds a new debugReport nested in the current debugReport, returning the nested report
-func (dr *debugReport) add(line string) *debugReport {
-	result := &debugReport{line: line}
+// Addf formats a new line in the report, returning a nested report for any additional information
+func (dr *StructureReport) Addf(format string, a ...any) *StructureReport {
+	result := &StructureReport{line: fmt.Sprintf(format, a...)}
 	dr.nested = append(dr.nested, result)
 	return result
 }
 
-func (dr *debugReport) saveTo(writer io.Writer) error {
+func (dr *StructureReport) SaveTo(writer io.Writer) error {
 	var indents []string
 
 	return dr.writeBlock(writer, indents, "", "")
 }
 
-// writeTo writes a block of lines from this debugReport to a writer
-func (dr *debugReport) writeBlock(
+// writeTo writes a block of lines from this StructureReport to a writer
+func (dr *StructureReport) writeBlock(
```

Comment:
minor: `sr` would make more sense than `dr` as the receiver name

Reply:
Fixed.